system_metrics_collector/src/system_metrics_collector/linux_process_memory_measurement_node.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "../../src/system_metrics_collector/linux_process_memory_measurement_node.hpp"
+
+#include <sys/sysinfo.h>
+#include <sys/types.h>
+
+#include <chrono>
+#include <cmath>
+#include <sstream>
+#include <string>
+
+#include "rcutils/logging_macros.h"
+
+namespace system_metrics_collector
+{
+
+LinuxProcessMemoryMeasurementNode::LinuxProcessMemoryMeasurementNode(
+  const std::string & name,
+  const std::chrono::milliseconds measurement_period,
+  const std::string & topic,
+  const std::chrono::milliseconds publish_period)
+: PeriodicMeasurementNode(name, measurement_period, topic, publish_period),
+  pid_(std::to_string(getPid())),
+  file_to_read_(PROC + pid_ + STATM)
+{
+}
+
+double LinuxProcessMemoryMeasurementNode::periodicMeasurement()
+{
+  const auto statm_line = readFileToString(file_to_read_);
+  const auto p_mem = getProcessUsedMemory(statm_line);
+  const auto total_mem = getSystemTotalMemory();
+
+  return p_mem / total_mem * 100.0;
+}
+
+std::string LinuxProcessMemoryMeasurementNode::getMetricName() const
+{
+  return pid_ + METRIC_NAME;
+}
+
+double getProcessUsedMemory(
```

Comment:
It seems really weird for this to be a double, we won't have 32.5 bytes of memory usage. Use uint64_t instead?

Reply:
That's a good point: I'll parse and return as uint64_t in this function to leave the casting responsibility to the user. It was done this way because ultimately the computation required a double.